common/src/ray_tracing.rs

```diff
@@ -1,13 +1,26 @@
-use crate::math;
+use crate::{
+    dodeca::Vertex,
+    math,
+    node::{Chunk, ChunkId, DualGraph, VoxelData},
+    world::Material,
+};
 
-/// Performs ray tracing against the z=0 plane in node coordinates. This plane is one-sided, so the ray must be going in the -z
-/// direction to detect an intersection.
+/// Performs ray tracing against the voxels in the `DualGraph`. This function is suitable for collision checking with
+/// any different collider shape. Each collider would have a different implemenation of `ChunkRayTracer`, which describes
+/// how the collision checking math works within a single chunk.
```

Comment:
Do we have immediate plans for any other shapes to be tested against voxels? If not, a trait seems very premature.

Reply:
We will want to test points for the ray-casting used to allow block placing and breaking.

Reply:
The use case already exists in the collision-demo branch. Any logic involving rays can benefit from the graph traversal algorithm in ray_tracing.rs.

However, I understand that the design can change far more than originally anticipated, so I'm okay with getting rid of this abstraction.

Reply:
It sounds like we'll eventually want to factor out the graph traversal algorithm into a more abstract swept broadphase, then. I'd definitely favor leaving the abstraction for that until we actually need it.

Reply:
"swept broadphase" is a terse description of the graph traversal algorithm.